sdk/eventhub/Azure.Messaging.EventHubs.Shared/tests/Core/ConnectionStringParserTests.cs

```diff
@@ -256,6 +257,37 @@ public void ParseIgnoresUnknownTokens()
             Assert.That(parsed.EventHubName, Is.EqualTo(eventHub), "The Event Hub path should match.");
         }
 
+        /// <summary>
+        ///   Verifies functionality of the <see cref="ConnectionStringParser.Parse" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase("test.endpoint.com")]
+        [TestCase("test.endpoint.com:443")]
+        [TestCase("sb://test.endpoint.com")]
+        [TestCase("sb://test.endpoint.com:80")]
+        [TestCase("amqp://test.endpoint.com")]
+        [TestCase("http://test.endpoint.com")]
+        [TestCase("https://test.endpoint.com:8443")]
+        public void ParseDoesAcceptsHostNamesAndUrisForTheEndpoint(string endpointValue)
+        {
+            var connectionString = $"Endpoint={ endpointValue };EntityPath=dummy";
+            var parsed = ConnectionStringParser.Parse(connectionString);
+
+            if (!Uri.TryCreate(endpointValue, UriKind.Absolute, out var valueUri))
+            {
+                valueUri = new Uri($"fake://{ endpointValue }");
+            }
+
+            Assert.That(parsed.Endpoint.Port, Is.EqualTo(-1), "The default port should be used.");
+            Assert.That(parsed.Endpoint.Host, Does.Not.Contain(" "), "The host name should not contain any spaces");
+            Assert.That(parsed.Endpoint.Host, Does.Not.Contain(":"), "The host name should not contain any port separators (:)");
+            Assert.That(parsed.Endpoint.Host, Does.Not.Contain(valueUri.Port), "The host name should not contain any spaces");
```

Comment:
I think this explanation string for the test is wrong.

Reply:
Yay, copy/paste.  Fixed.